solcx/install.py

```diff
@@ -226,23 +237,26 @@ def install_solc(version, allow_osx=False, show_progress=False, solcx_binary_pat
             return
         return install_solc(version, allow_osx)
 
-    try:
-        if platform == "linux":
-            _install_solc_linux(version, show_progress, solcx_binary_path)
-        elif platform == "darwin":
-            _install_solc_osx(version, allow_osx, show_progress, solcx_binary_path)
-        elif platform == "win32":
-            _install_solc_windows(version, show_progress, solcx_binary_path)
-        binary_path = get_executable(version, solcx_binary_path)
-        _check_subprocess_call(
-            [binary_path, "--version"],
-            message="Checking installed executable version @ {}".format(binary_path),
-        )
-        if not solc_version:
-            set_solc_version(version)
-        LOGGER.info("solc {} successfully installed at: {}".format(version, binary_path))
-    finally:
-        lock.release()
+    if arch == "arm":
```

Comment:
I think this would be more fitting as the first `if` statement in the sequence of `if`/`elif` starting at line 244.  It's important that it happens within the context of `try`/`finally` so that the process lock is released when it completes.